src/libraries/System.Net.Http/tests/FunctionalTests/NtAuthTests.FakeServer.cs

```diff
@@ -0,0 +1,144 @@
+            }
+
+            bool isAuthenticated = false;
+            do
+            {
+                foreach (HttpHeaderData header in request.Headers)
+                {
+                    if (header.Name == "Authorization")
+                    {
+                        authHeader = header.Value;
+                        break;
+                    }
+                }
+
+                Assert.NotNull(authHeader);
+                var tokens = authHeader.Split(' ', 2, StringSplitOptions.TrimEntries);
+                // Should be type and base64 encoded blob
+                Assert.Equal(2, tokens.Length);
+
+                if (fakeNtlmServer == null)
+                {
+                    fakeNtlmServer = new FakeNtlmServer(s_testCredentialRight) { ForceNegotiateVersion = true };
+                    if (!useNtlm)
+                    {
+                        fakeNegotiateServer = new FakeNegotiateServer(fakeNtlmServer);
+                    }
+                }
+
+                byte[]? outBlob;
+
+                if (fakeNegotiateServer != null)
+                {
+                    outBlob = fakeNegotiateServer.GetOutgoingBlob(Convert.FromBase64String(tokens[1]));
+                    isAuthenticated = fakeNegotiateServer.IsAuthenticated;
+                }
+                else
+                {
+                    outBlob = fakeNtlmServer.GetOutgoingBlob(Convert.FromBase64String(tokens[1]));
+                    isAuthenticated = fakeNtlmServer.IsAuthenticated;
+                }
+
+                if (outBlob != null)
+                {
+                    authHeader = $"WWW-Authenticate: {tokens[0]} {Convert.ToBase64String(outBlob)}\r\n";
+                    await connection.SendResponseAsync(isAuthenticated ? HttpStatusCode.OK : HttpStatusCode.Unauthorized, authHeader);
+                    connection.CompleteRequestProcessing();
+
+                    if (!isAuthenticated)
+                    {
+                        request = await connection.ReadRequestDataAsync();
+                    }
+                }
+            }
+            while (!isAuthenticated);
+
+            await connection.SendResponseAsync(HttpStatusCode.OK);
+        }
+
+        [ConditionalTheory(nameof(IsNtlmInstalled))]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public async Task SendAsync_FakeServer_Success(bool useDefaultHandler, bool useNtlm)
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Get, uri);
+                    requestMessage.Version = new Version(1, 1);
+
+                    HttpMessageHandler handler = useDefaultHandler ? 
+                        new HttpClientHandler() { Credentials = s_testCredentialRight } :
```

Comment:
cc @simonrozsival

We should test that this code path works in the unit tests and goes through the Android handler once all the pieces are in place.

Reply:
I'm not sure if this test runs on Android at all. I think `Capability.IsNtlmInstalled()` will return `false` because it looks for GSS and that should fail. I test it later.

Reply:
Fair point, I will remove the condition.